x/storage/keeper/keeper.go

```diff
@@ -230,31 +231,46 @@ func (k Keeper) ForceDeleteBucket(ctx sdk.Context, bucketId sdkmath.Uint, cap ui
 		var objectInfo types.ObjectInfo
 		k.cdc.MustUnmarshal(bz, &objectInfo)
 
-		if objectInfo.ObjectStatus == types.OBJECT_STATUS_CREATED {
-			if err := k.UnlockStoreFee(ctx, bucketInfo, &objectInfo); err != nil {
+		// An object cannot be discontinued if the bucket is already discontinued,
+		// which means that after deleting objects when deleting a bucket the objects in it should be in
+		// OBJECT_STATUS_CREATED or OBJECT_STATUS_SEALED status.
+		// However, when updating the deletion confirm period parameter, it can lead to that
+		// the discontinued bucket can be deleted earlier, then its objects could be in
+		// OBJECT_STATUS_DISCONTINUED status.
+
+		objectStatus := objectInfo.ObjectStatus
+		if objectStatus == types.OBJECT_STATUS_DISCONTINUED {
+			objectStatus, err = k.getDiscontinueObjectStatus(ctx, objectInfo.Id)
+			if err != nil {
```

Comment:
If an error occurs, does the object need to wait for the `discontinueObject` to be deleted?